### DISCUSSION TITLE
zenoh pico arduino framework on esp32

### DISCUSSION THREAD

**Discussion by robobe** [eclipse-zenoh/roadmap#166] [Q&A]
Hi
I am try to build pico for arduino framework on esp32 platform using platformio, when i build for the espidf framework everything ok and it work
when i build for arduino i got an error "#error "Multi-thread refcount in C99 only exists for GCC, use GCC or C11 or deactivate multi-thread" what the correct flags to success compiling

[env:esp32dev]
platform = espressif32
board = esp32dev
framework = arduino
; framework = espidf
upload_port = /dev/ttyUSB1
monitor_port= /dev/ttyUSB1
monitor_speed = 115200
lib_deps = https://github.com/eclipse-zenoh/zenoh-pico

build_flags =-D ZENOH_ARDUINO_ESP32=1

**Comment by artem30801** [comment]
Hello! I was able to build with flags like these:

```
[env:esp32-s3-devkitc-1]
platform = espressif32
board = esp32-s3-devkitc-1
framework = arduino
build_flags =
    -DZENOH_COMPILER_GCC
    -DZENOH_ARDUINO_ESP32
lib_deps = https://github.com/eclipse-zenoh/zenoh-pico
```

**Reply by BenCaunt** [reply]
thank you so much
